### DISCUSSION TITLE
ENABLING NET MQTT LOG DEBUG

### DISCUSSION THREAD

**Discussion by zafersn** [zephyrproject-rtos/zephyr#64821] [Q&A] [answered]
I am trying to enable debug logs for the mqtt 

so to see debug messages for the mqtt: I used:

CONFIG_NET_LOG=y
CONFIG_MQTT_LOG_LEVEL_DBG=y

here is the log module reg. for mqtt in the mqtt.c :  

LOG_MODULE_REGISTER(net_mqtt, CONFIG_MQTT_LOG_LEVEL);

What configs else should I use to be able to see debug prints?

**Comment by jukkar** [comment]
What you have set below is enough.
```
CONFIG_NET_LOG=y
CONFIG_MQTT_LOG_LEVEL_DBG=y
```
Of course you need to enable MQTT too with `CONFIG_MQTT_LIB=y`.

**Reply by zafersn** [reply]
But it is not, I cant see the logs (NET logs in mqtt lib) can you please verify if it is ok
using zephyr 3.5.0

**Reply by jukkar** [reply, marked as answer]
The following worked with latest upstream:
```
west build -b native_posix -d build/mqtt-sample zephyr/samples/net/mqtt_publisher -- -DCONFIG_NET_SHELL=y -DCONFIG_MQTT_LOG_LEVEL_DBG=y -DCONFIG_NET_LOG=y -DCONFIG_NATIVE_UART_0_ON_STDINOUT=y
./build/mqtt-sample/zephyr/zephyr.exe
```
and then I am seeing
```
[00:00:00.120,000] <inf> net_mqtt_publisher_sample: mqtt_connect: -110 <ERROR>
[00:00:00.630,000] <dbg> net_mqtt_sock_tcp: mqtt_client_tcp_connect: (main): Created socket 0
```
where the mqtt library prints debug messages. There should be no difference between 3.5 and current development head in this respect.
